tools/windows/install-help/cal/FinalizeInstall.cpp

```diff
@@ -96,20 +97,17 @@ UINT doFinalizeInstall(CustomActionData &data)
         }
     }
 
-    // add all the rights we want to the user (either existing or newly created)
+    auto sid = data.Sid();
```

Comment:
this works, but kinda by accident.  It only works because the scope of `data` is larger than the scope of `sid`.  But, it's asking for bugs in the future.  If some code modifies `data.sid_ptr`, this variable will be left dangling, pointing to now invalidated memory.

if you want to go down this path, then data.sid_ptr should probably be a `std::shared_ptr`.  `data.Sid()` can the return a new instance of the `shared_ptr()`, and it will be safe from being changed out from under it.